diff --git a/src/nodes/core/Node.d.ts b/src/nodes/core/Node.d.ts
index 162515f6d8467c5139edded4def574ad988e7c44..d03d3c0f8eb0bf6d7802ba20ec13e09de80d02ae 100644
--- a/src/nodes/core/Node.d.ts
+++ b/src/nodes/core/Node.d.ts
@@ -556,63 +556,60 @@ export interface Mat4Extensions {
 export interface MatExtensions<TMat extends MatType> {
 }
 
+interface NodeExtras {
+    [x: string]: {};
+    "float": NumOrBoolExtensions<"float"> & FloatExtensions & NumExtensions<"float"> & FloatOrVecExtensions<"float">;
+    "int":
+        & NumOrBoolExtensions<"int">
+        & IntExtensions
+        & NumExtensions<"int">
+        & IntegerExtensions<"int">
+        & IntOrVecExtensions<"int">;
+    "uint":
+        & NumOrBoolExtensions<"uint">
+        & UintExtensions
+        & NumExtensions<"uint">
+        & IntegerExtensions<"uint">
+        & IntOrVecExtensions<"uint">;
+    "bool": NumOrBoolExtensions<"bool"> & BoolExtensions & BoolOrVecExtensions<"bool">;
+    "vec2":
+        & NumOrBoolVec2Extensions<"float">
+        & Vec2Extensions
+        & NumVec2Extensions<"float">
+        & FloatVecExtensions<"vec2">
+        & FloatOrVecExtensions<"vec2">;
+    "ivec2": NumOrBoolVec2Extensions<"int"> & NumVec2Extensions<"int"> & IntOrVecExtensions<"ivec2">;
+    "uvec2": NumOrBoolVec2Extensions<"uint"> & NumVec2Extensions<"uint"> & IntOrVecExtensions<"uvec2">;
+    "bvec2": NumOrBoolVec2Extensions<"bool"> & BoolOrVecExtensions<"bvec2">;
+    "vec3":
+        & NumOrBoolVec3Extensions<"float">
+        & Vec3Extensions
+        & NumVec3Extensions<"float">
+        & FloatVecExtensions<"vec3">
+        & FloatOrVecExtensions<"vec3">;
+    "ivec3": NumOrBoolVec3Extensions<"int"> & NumVec3Extensions<"int"> & IntOrVecExtensions<"ivec3">;
+    "uvec3": NumOrBoolVec3Extensions<"uint"> & NumVec3Extensions<"uint"> & IntOrVecExtensions<"uvec3">;
+    "bvec3": NumOrBoolVec3Extensions<"bool"> & BoolOrVecExtensions<"bvec3">;
+    "vec4":
+        & NumOrBoolVec4Extensions<"float">
+        & Vec4Extensions
+        & NumVec4Extensions<"float">
+        & FloatVecExtensions<"vec4">
+        & FloatOrVecExtensions<"vec4">;
+    "ivec4": NumOrBoolVec4Extensions<"int"> & NumVec4Extensions<"int"> & IntOrVecExtensions<"ivec4">;
+    "uvec4": NumOrBoolVec4Extensions<"uint"> & NumVec4Extensions<"uint"> & IntOrVecExtensions<"uvec4">;
+    "bvec4": NumOrBoolVec4Extensions<"bool"> & BoolOrVecExtensions<"bvec4">;
+    "color": ColorExtensions;
+    "mat2": Mat2Extensions & MatExtensions<"mat2">;
+    "mat3": Mat3Extensions & MatExtensions<"mat3">;
+    "mat4": Mat4Extensions & MatExtensions<"mat4">;
+}
+
 type Node<TNodeType = unknown> =
     & NodeClass
     & NodeElements
     & (unknown extends TNodeType ? {} : NodeExtensions<TNodeType>)
-    & (TNodeType extends "float"
-        ? NumOrBoolExtensions<"float"> & FloatExtensions & NumExtensions<"float"> & FloatOrVecExtensions<"float">
-        : TNodeType extends "int" ?
-                & NumOrBoolExtensions<"int">
-                & IntExtensions
-                & NumExtensions<"int">
-                & IntegerExtensions<"int">
-                & IntOrVecExtensions<"int">
-        : TNodeType extends "uint" ?
-                & NumOrBoolExtensions<"uint">
-                & UintExtensions
-                & NumExtensions<"uint">
-                & IntegerExtensions<"uint">
-                & IntOrVecExtensions<"uint">
-        : TNodeType extends "bool" ? NumOrBoolExtensions<"bool"> & BoolExtensions & BoolOrVecExtensions<"bool">
-        : TNodeType extends "vec2" ?
-                & NumOrBoolVec2Extensions<"float">
-                & Vec2Extensions
-                & NumVec2Extensions<"float">
-                & FloatVecExtensions<"vec2">
-                & FloatOrVecExtensions<"vec2">
-        : TNodeType extends "ivec2"
-            ? NumOrBoolVec2Extensions<"int"> & NumVec2Extensions<"int"> & IntOrVecExtensions<"ivec2">
-        : TNodeType extends "uvec2"
-            ? NumOrBoolVec2Extensions<"uint"> & NumVec2Extensions<"uint"> & IntOrVecExtensions<"uvec2">
-        : TNodeType extends "bvec2" ? NumOrBoolVec2Extensions<"bool"> & BoolOrVecExtensions<"bvec2">
-        : TNodeType extends "vec3" ?
-                & NumOrBoolVec3Extensions<"float">
-                & Vec3Extensions
-                & NumVec3Extensions<"float">
-                & FloatVecExtensions<"vec3">
-                & FloatOrVecExtensions<"vec3">
-        : TNodeType extends "ivec3"
-            ? NumOrBoolVec3Extensions<"int"> & NumVec3Extensions<"int"> & IntOrVecExtensions<"ivec3">
-        : TNodeType extends "uvec3"
-            ? NumOrBoolVec3Extensions<"uint"> & NumVec3Extensions<"uint"> & IntOrVecExtensions<"uvec3">
-        : TNodeType extends "bvec3" ? NumOrBoolVec3Extensions<"bool"> & BoolOrVecExtensions<"bvec3">
-        : TNodeType extends "vec4" ?
-                & NumOrBoolVec4Extensions<"float">
-                & Vec4Extensions
-                & NumVec4Extensions<"float">
-                & FloatVecExtensions<"vec4">
-                & FloatOrVecExtensions<"vec4">
-        : TNodeType extends "ivec4"
-            ? NumOrBoolVec4Extensions<"int"> & NumVec4Extensions<"int"> & IntOrVecExtensions<"ivec4">
-        : TNodeType extends "uvec4"
-            ? NumOrBoolVec4Extensions<"uint"> & NumVec4Extensions<"uint"> & IntOrVecExtensions<"uvec4">
-        : TNodeType extends "bvec4" ? NumOrBoolVec4Extensions<"bool"> & BoolOrVecExtensions<"bvec4">
-        : TNodeType extends "color" ? ColorExtensions
-        : TNodeType extends "mat2" ? Mat2Extensions & MatExtensions<"mat2">
-        : TNodeType extends "mat3" ? Mat3Extensions & MatExtensions<"mat3">
-        : TNodeType extends "mat4" ? Mat4Extensions & MatExtensions<"mat4">
-        : {})
+    & NodeExtras[TNodeType & string]
     & {
         __TypeScript_NODE_TYPE__: TNodeType;
     };
